src/commonMain/kotlin/de/voize/semver4k/Requirement.kt

```diff
@@ -87,6 +87,61 @@ class Requirement
         }
     }
 
+    fun minVersion(): Semver? {
+        var minver: Semver? = Semver("0.0.0")
+        if (minver != null && this.isSatisfiedBy(minver)) {
+            return minver
+        }
+
+        minver = Semver("0.0.0-0")
+        if (this.isSatisfiedBy(minver)) {
+            return minver
+        }
+
+        minver = null
+
+        val ranges = mutableListOf<Range?>()
+        this.getAllAllRanges(this, ranges)
+
+        for (range in ranges) {
+            if (range == null) {
+                continue;
+            }
+
+            var compver = range.version;
+            if (range.op.equals(RangeOperator.GT)) {
+                compver = compver.nextIncrement();
+            }
+
+            // Pick this new version if it's smaller than the one we already have and still satisfies the requirements
+            if ((range.op.equals(RangeOperator.GT)
+                || range.op.equals(RangeOperator.GTE)
+                || range.op.equals(RangeOperator.EQ))
+                && (minver == null || minver.compareTo(compver) > 0)
+                && this.isSatisfiedBy(compver)
+            ) {
+                minver = compver
+            }
+        }
+
+        if (minver != null && this.isSatisfiedBy(minver)) {
+            return minver
+        }
+
+        return null
+    }
+
+    private fun getAllAllRanges(requirement: Requirement?, res: MutableList<Range?>): List<Range?> {
```

Comment:
maybe give it another name

Reply:
oops, right, I forgot to rename that after experimenting, I will do that

Reply:
done, renamed to `getAllGreaterOrEqualRanges`